test/show.jl

```diff
@@ -1922,7 +1922,7 @@ Z = Array{Float64}(undef,0,0)
     @test showstr(vec_undefined) == "Any[#undef, #undef]"
     @test showstr(vec_initialisers) == "[$undef, $undef]"
     @test replstr(vec_undefined) == "2-element Array{Any,1}:\n #undef\n #undef"
-    @test replstr(vec_initialisers) == "2-element Array{UndefInitializer,1}:\n $undef\n $undef"
+    @test replstr(vec_initialisers) == "2-element Array{UndefInitializer,1}:\n UndefInitializer(): array initializer with undefined values\n UndefInitializer(): array initializer with undefined values"
```

Comment:
The compact form would be better in this case. I'm. It sure how that can be done without impacting the 3-arg show used on the REPL

Reply:
Agreed; after this is merged we can add some handling for `:compact` in the 3-arg show methods for UndefInitializer and Char and maybe some others.